### ISSUE TITLE
Rebuild od2 cluster

### ISSUE THREAD

**Issue by decimalator** [OregonDigital/OD2#935]
### Descriptive summary

There are some storage limitations with our current cluster and a new version of k8s we can upgrade to, so I am going to build up a new cluster and move us over to it.

There should be minimal impact to staging, I can bring the new cluster up and attach to the existing storage for current deployments once it's ready to bring up.

**Comment by DeadCatLady** [comment]
This should fix the evicted pods that are running out of ephemeral storage.

**Comment by decimalator** [comment]
The new cluster is up and running. We have integrated it with our Rancher cluster so everyone will be able to access both clusters. We will work with developers to get them setup in the new cluster and get services and data migrated over next week.

**Comment by decimalator** [comment]
The `od2020` cluster is done and waiting for QA. The following should be tested to verify the new cluster works well enough that we can spin down the old `od2` cluster.

This is effectively both a cluster test and a disaster recovery test since I attempted to migrate a partially consistent set of backups from Fedora, SOLR, Postgres and Redis to restore what should be the same service along with its data on a new set of hardware.

### Postgres
- [ ] Postgres is not missing data
- [ ] Operations that require database table updates should be tested to make sure they work

We should do some basic data consistency tests, although in all cases we've had over a week for content drift so we'll need to ignore new data that has been ingested into `od2` since the snapshots that were created to bring up `od2020`

### Fedora
- [ ] Fedora is not missing data
- [ ] Verify we can create, modify and delete objects from Fedora

The same goes for Fedora. It should be easy enough to walk through the repo to compare.

### SOLR
- [ ] SOLR is not missing data
- [ ] Verify we can create, modify and delete documents in SOLR

The same goes for SOLR. We need to put together some SOLR queries to test that the data that's there is everything that came from the other SOLR, minus the new data of course.

### Redis
- [ ] Redis is not missing data
- [ ] The biggest test for Redis is that workers jobs run, but I think there were some other things stored in Redis too?

Same for redis, although @jechols has said in the past that there are some other, less simple things in Redis too so we'll need to make sure those don't break.

**Comment by jechols** [comment]
I'm not sure how to address the situation of content drift.  In a normal backup / restore test, we could check various things before the backup and validate that they're the same after the restore.  It's in that vein that I can respond to some of this in a general case, but for this specific situation... I'm not sure how to verify much beyond that the things "seem good enough".  e.g., the site has assets and the search returns some results.

I think the redis data is a mystery to everybody, unfortunately.  I would have to do some in-depth digging in the stack to figure out what's there and how to verify it.

For Fedora, there are APIs we can hit to verify simple things like just getting a full list of ids.  My WIP reindex task basically does exactly that to determine what to send off to reindex jobs.  We could even pull the content from those ids to verify things like "there are X Image objects".  This could be automated pretty trivially, but I think it'll take some discussion to determine what we want to test.

Solr is even easier since pulling the data is so much faster than Fedora.  We can just pull whatever we want - list of Image assets' ids, for instance - and verify whatever we want.

We could also verify that the Fedora object ids match the solr ids.  This is a tricky one, though - I think things get orphaned sometimes, and it's not in our control (I discovered at least one case where Fedora can have objects which are not in Solr).  But we could verify the "important" data is in sync, e.g., everything that's an Image / Document / Audio / etc.  That way we don't see an error if the stack is doing something weird we can't control.  We just care that the visible assets are synced between Fedora and Solr.

**Comment by lsat12357** [comment]
If by not being able to do a full analysis on the data, we are holding up the switchover to the new cluster, I think we need to separate that out and have it be its own issue, so we can come up with a reasonable way to compare data sets.

**Comment by lsat12357** [comment]
I checked off create/modify/delete for Fedora and Solr based on migration testing. For Redis, the migration required that several jobs be launched, and these appear to have completed successfully; have not looked into what other things Redis stores.

**Comment by lsat12357** [comment]
@jsimic volunteered to help with testing through the front end but is not available until next week, and has requested a checklist.

**Comment by decimalator** [comment]
I just found this comment unfinished in a tab on my other computer 🤦‍♀ 

awesome thanks for the feedback!

we'd definitely want to do a test at some point that's more exhaustive so we know that we can move this data around and know it's going to be there. maybe some kind of data validation thing that can check the number of items in fedora, the number in solr and give us a side-by-side high level overview that we can easily compare. and then a more detailed dump of each individual item in some format that makes it easy to compare 

for the front end, I think Features would know the various things to test so that we're exercising all of the right things. who can put together a web UI testing checklist?

**Comment by raybrarian** [comment]
To get Migration Team unblocked, I'm considering this one essentially a QA pass, and @shieldsb can continue to report any discrepancies through post-move GH issues.
